api/v2_recommendations.go

```diff
@@ -0,0 +1,144 @@
+	Patch(recommendations RecommendationStateV2) (RecommendationResponseV2, error)
+	GetReport(reportType string) ([]RecV2, error)
+}
+
+type RecommendationTypeV2 string
+
+const (
+	AwsRecommendation   RecommendationTypeV2 = "aws"
+	AzureRecommendation RecommendationTypeV2 = "azure"
+	GcpRecommendation   RecommendationTypeV2 = "gcp"
+)
+
+func (svc *RecommendationsServiceV2) list(cloudType RecommendationTypeV2) ([]RecV2, error) {
+	var response RecommendationResponseV2
+	err := svc.client.RequestDecoder("GET", fmt.Sprintf(apiRecommendations, cloudType), nil, &response)
+	return response.RecommendationList(), err
+}
+
+func (svc *RecommendationsServiceV2) patch(cloudType RecommendationTypeV2, recommendations RecommendationStateV2) (
+	response RecommendationResponseV2,
+	err error,
+) {
+	err = svc.client.RequestEncoderDecoder("PATCH", fmt.Sprintf(apiRecommendations, cloudType), recommendations, &response)
+	return
+}
+
+type RecommendationStateV2 map[string]string
+
+type RecommendationDataV2 map[string]RecommendationEnabledV2
+
+type RecV2 struct {
+	ID    string
+	State bool
+}
+
+type RecommendationEnabledV2 struct {
+	Enabled bool `json:"enabled"`
+}
+
+type RecommendationResponseV2 struct {
+	Data    []RecommendationDataV2 `json:"data"`
+	Ok      bool                   `json:"ok"`
+	Message string                 `json:"message"`
+}
+
+func (res *RecommendationResponseV2) RecommendationList() (recommendations []RecV2) {
+	if len(res.Data) > 0 {
+		for k, v := range res.Data[0] {
+			recommendations = append(recommendations, RecV2{k, v.Enabled})
+		}
+	}
+	return
+}
+
+type ReportSchema struct {
+	Name              string   `json:"name"`
+	RecommendationIDs []string `json:"recommendationIDs"`
+}
+
+func NewRecommendationV2State(recommendations []RecV2, state bool) RecommendationStateV2 {
+	request := make(map[string]string)
+	for _, rec := range recommendations {
+		if state {
+			request[rec.ID] = "enable"
+
+		} else {
+			request[rec.ID] = "disable"
+		}
+	}
+	return request
+}
+
+func NewRecommendationV2(recommendations []RecV2) RecommendationStateV2 {
+	request := make(map[string]string)
+	for _, rec := range recommendations {
+		if rec.State {
+			request[rec.ID] = "enable"
+
+		} else {
+			request[rec.ID] = "disable"
+		}
+	}
+	return request
+}
+
+// ReportStatus This is an experimental feature. Returned RecommendationID's are not guaranteed to be correct.
+func (res *RecommendationResponseV2) ReportStatus() map[string]bool {
+	var recommendations = make(map[string]bool)
+
+	for _, rec := range res.RecommendationList() {
+		recommendations[rec.ID] = rec.State
+	}
+
+	return recommendations
+}
+
+// filterRecommendations This is an experimental feature. Returned RecommendationID's are not guaranteed to be correct.
+func filterRecommendations(allRecommendations []RecV2, schema ReportSchema) []RecV2 {
+	var recommendations []RecV2
+
+	for _, rec := range allRecommendations {
+		if array.ContainsStr(schema.RecommendationIDs, rec.ID) {
```

Comment:
You could look at changing allRecommendations to a map to avoid the nested loop.
Worth noting this only runs once per report to build the report type mappings and then it's cached.